deploy/crds/openebs.io_blockdeviceclaims_crd.yaml

```diff
@@ -0,0 +1,157 @@
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: DeviceClaimSpec defines the request details for a BlockDevice
+            properties:
+              blockDeviceName:
+                description: BlockDeviceName is the reference to the block-device
+                  backing this claim
+                type: string
+              blockDeviceNodeAttributes:
+                description: BlockDeviceNodeAttributes is the attributes on the node
+                  from which a BD should be selected for this claim. It can include
+                  nodename, failure domain etc.
+                properties:
+                  hostName:
+                    description: HostName represents the hostname of the Kubernetes
+                      node resource where the BD should be present
+                    type: string
+                  nodeName:
+                    description: NodeName represents the name of the Kubernetes node
+                      resource where the BD should be present
+                    type: string
+                type: object
+              deviceClaimDetails:
+                description: Details of the device to be claimed
+                properties:
+                  allowPartition:
+                    description: AllowPartition represents whether to claim a full
+                      block device or a device that is a partition
+                    type: boolean
+                  blockVolumeMode:
+                    description: 'BlockVolumeMode represents whether to claim a device
+                      in Block mode or Filesystem mode. These are use cases of BlockVolumeMode:
+                      1) Not specified: VolumeMode check will not be effective 2)
+                      VolumeModeBlock: BD should not have any filesystem or mountpoint
+                      3) VolumeModeFileSystem: BD should have a filesystem and mountpoint.
+                      If DeviceFormat is    specified then the format should match
+                      with the FSType in BD'
+                    type: string
+                  formatType:
+                    description: Format of the device required, eg:ext4, xfs
+                    type: string
+                type: object
+              deviceType:
+                description: DeviceType represents the type of drive like SSD, HDD
+                  etc.,
+                type: string
+              hostName:
+                description: Node name from where blockdevice has to be claimed. To
+                  be deprecated. Use NodeAttributes.HostName instead
+                type: string
+              resources:
+                description: Resources will help with placing claims on Capacity,
+                  IOPS
+                properties:
+                  requests:
```

Comment:
@harshthakur9030 . The following BDC still does not work.
```yaml
apiVersion: openebs.io/v1alpha1
kind: BlockDeviceClaim
metadata:
  name: bdc-ssd
spec:
  blockDeviceNodeAttributes:
    hostName: worker-ak1
  resources:
    requests:
    - storage: 1Gi 
```

with error
```
The BlockDeviceClaim "bdc-ssd" is invalid: spec.resources.requests: Invalid value: "array": spec.resources.requests in body must be of type object: "array"
```

Reply:
This is confusing. Any idea how to fix this?

Reply:
i think its indentation problem it should be like this ?
```  yaml
resources:
    requests:
      storage: 1Gi
```

Reply:
Sorry, It was en error in the yaml I had. Can ignore this comment.